flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/source/assigner/SimpleSplitAssigner.java

```diff
@@ -51,7 +51,7 @@ public SimpleSplitAssigner(Collection<IcebergSourceSplitState> assignerState) {
   }
 
   @Override
-  public GetSplitResult getNext(@Nullable String hostname) {
+  public synchronized GetSplitResult getNext(@Nullable String hostname) {
```

Comment:
added `synchronized` because the new `pendingSplitCount` method can be called from I/O threads, while other methods are only called from the coordinator thread.